### PR TITLE
Pharmacy - Change, Limiting non-opioid relief and expanding opioid effectiveness 

### PR DESCRIPTION
**Pull request by mazinskihenry**
**When merged this pull request will:**
- Create a maxRelief variable for medications to create soft ceilings for pain reduction 
- Apply this ceiling to Painkillers and Penthrox 
- Adding relief factors to opioids which will make users more "tankier", or at least perceive it that way

### IMPORTANT

- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/pharma/functions/fnc_treatmentAdvanced_FentanylLocal.sqf [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
 
-
-/// ChromAberration effect
-params ["_target"];
+params ["_target", "_bodyPart", "_opioidRelief"];
 
 if !(alive _target) exitWith {};
 if (ACE_Player != _target) exitWith {};
 
+private _opioidFactor = _target getVariable (QGVAR(opioidFactor), 0);
```
Comment (MiszczuZPolski) [inline addons/pharma/functions/fnc_treatmentAdvanced_FentanylLocal.sqf]:
```suggestion
private _opioidFactor = _target getVariable [QGVAR(opioidFactor), 0];
```

File: addons/pharma/functions/fnc_treatmentAdvanced_FentanylLocal.sqf [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
+if (_opioidFactor == 1) then {
+    _target setVariable (QGVAR(opioidFactor), _opioidRelief, true);
```
Comment (MiszczuZPolski) [inline addons/pharma/functions/fnc_treatmentAdvanced_FentanylLocal.sqf]:
```suggestion
    _target setVariable [QGVAR(opioidFactor), _opioidRelief, true];
```

File: addons/pharma/functions/fnc_treatmentAdvanced_MorphineLocal.sqf [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Public: No
+ */
+
+params ["_patient", "_bodyPart", "_opioidRelief"];
+
+if !(alive _patient) exitWith {};
+if (ACE_Player != _patient) exitWith {};
+
+private _opioidFactor = _patient getVariable (QGVAR(opioidFactor), 0);
```
Comment (MiszczuZPolski) [inline addons/pharma/functions/fnc_treatmentAdvanced_MorphineLocal.sqf]:
```suggestion
private _opioidFactor = _patient getVariable [QGVAR(opioidFactor), 0];
```

File: addons/pharma/functions/fnc_treatmentAdvanced_MorphineLocal.sqf [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+if (_opioidFactor == 1) then {
+    _patient setVariable (QGVAR(opioidFactor), _opioidRelief, true);
```
Comment (MiszczuZPolski) [inline addons/pharma/functions/fnc_treatmentAdvanced_MorphineLocal.sqf]:
```suggestion
    _patient setVariable [QGVAR(opioidFactor), _opioidRelief, true];
```

File: addons/pharma/functions/fnc_treatmentAdvanced_NalbuphineLocal.sqf [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Public: No
+ */
+
+params ["_patient", "_bodyPart", "_opioidRelief"];
+
+if !(alive _patient) exitWith {};
+if (ACE_Player != _patient) exitWith {};
+
+private _opioidFactor = _patient getVariable (QGVAR(opioidFactor), 0);
```
Comment (MiszczuZPolski) [inline addons/pharma/functions/fnc_treatmentAdvanced_NalbuphineLocal.sqf]:
```suggestion
private _opioidFactor = _patient getVariable [QGVAR(opioidFactor), 0];
```

File: addons/pharma/functions/fnc_treatmentAdvanced_NalbuphineLocal.sqf [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+if (_opioidFactor == 1) then {
+    _patient setVariable (QGVAR(opioidFactor), _opioidRelief, true);
```
Comment (MiszczuZPolski) [inline addons/pharma/functions/fnc_treatmentAdvanced_NalbuphineLocal.sqf]:
```suggestion
    _patient setVariable [QGVAR(opioidFactor), _opioidRelief, true];
```

File: addons/pharma/functions/fnc_medicationLocal.sqf [hunk 7]
```diff
@@ -77,12 +77,23 @@ private _hrIncreaseNormal       = GET_ARRAY(_medicationConfig >> "hrIncreaseNorm
...
 private _alphaFactor            = GET_NUMBER(_medicationConfig >> "alphaFactor",getNumber (_defaultConfig >> "alphaFactor"));
+private _maxRelief              = GET_NUMBER(_medicationConfig >> "maxRelief",getNumber (_defaultConfig >> "maxRelief"));
+private _opioidRelief           = GET_NUMBER(_medicationConfig >> "opioidRelief",getNumber (_defaultConfig >> "opioidRelief"));
 
 private _heartRate = GET_HEART_RATE(_patient);
 private _hrIncrease = [_hrIncreaseLow, _hrIncreaseNormal, _hrIncreaseHigh] select (floor ((0 max _heartRate min 110) / 55));
 _hrIncrease params ["_minIncrease", "_maxIncrease"];
 private _heartRateChange = _minIncrease + random (_maxIncrease - _minIncrease);
 
+private _adjustments = _patient getVariable [VAR_MEDICATIONS,[]];
```
Comment (MiszczuZPolski) [inline addons/pharma/functions/fnc_medicationLocal.sqf]: where is it used?

File: addons/pharma/functions/fnc_medicationLocal.sqf [hunk 8]
```diff
@@ -77,12 +77,23 @@ private _hrIncreaseNormal       = GET_ARRAY(_medicationConfig >> "hrIncreaseNorm
...
+private _presentPain = GET_PAIN(_patient)
+private _presentReduce = 0;
+if (_maxRelief > 0) then {
+    if (_presentPain > _maxRelief) then {
+        _painReduce = _painReduce / 4;
```
Comment (MiszczuZPolski) [inline addons/pharma/functions/fnc_medicationLocal.sqf:92]: so if we have pain 1.0, and max Relief is only 0.6 then we will only lower it by only 0.25? looking at penthrox

File: addons/main/script_macros.hpp [hunk 9]
```diff
@@ -170,7 +171,7 @@
 #define GET_SM_STATE(_unit)         ([_unit, ACEGVAR(medical,STATE_MACHINE)] call CBA_statemachine_fnc_getCurrentState)
 
 #undef GET_DAMAGE_THRESHOLD
-#define GET_DAMAGE_THRESHOLD(unit)  (unit getVariable [QACEGVAR(medical,damageThreshold), [ACEGVAR(medical,AIDamageThreshold),ACEGVAR(medical,playerDamageThreshold)] select (isPlayer unit)])
+#define GET_DAMAGE_THRESHOLD(unit)  ((unit getVariable [QACEGVAR(medical,damageThreshold), [ACEGVAR(medical,AIDamageThreshold),ACEGVAR(medical,playerDamageThreshold)] select (isPlayer unit)]) * VAR_OPIOID_FACTOR)
```
Comment (MiszczuZPolski) [inline addons/main/script_macros.hpp:174]: 🤔 but opiod is never bigger than 1 so if i understand correctly we lower player damage threshold when we give them certain meds?

**Review by MiszczuZPolski — CHANGES_REQUESTED** [review]
Whats default value of opioid relief because in handleRespawn/init its 1 but in getVariable you used 0 as default value.
I would like to add more stuff about changes to your PR description so it can be undestood better

File: addons/pharma/functions/fnc_treatmentAdvanced_FentanylLocal.sqf [hunk 10]
```diff
@@ -1,27 +1,34 @@
...
  * 0: Patient <OBJECT>
+ * 1: Bodypart <STRING>
+ * 2: OpioidRelief <NUMBER>
  *
  * Return Value:
  * None
  *
  * Example:
- * [player] call kat_pharma_fnc_fentanylPP;
+ * [player, "LeftLeg", opioidRelief] call kat_pharma_fnc_treatmentAdvanced_FentanylLocal;
```
Comment (MissHeda) [inline addons/pharma/functions/fnc_treatmentAdvanced_FentanylLocal.sqf]: change opioidRelief to number in example call

File: addons/pharma/ACE_Medical_Treatment.hpp [hunk 11]
```diff
@@ -248,11 +288,12 @@ class ACE_ADDON(Medical_Treatment) {
...
             hrIncreaseHigh[] = {-5, -15};
             timeInSystem = 300;
-            timeTillMaxEffect = 10;
-            maxDose = 10;
+            timeTillMaxEffect = 20;
+            maxDose = 4;
             incompatibleMedication[] = {};
             viscosityChange = 5;
             onOverDose = "";
+            maxRelief = 0.6;
```
Comment (MissHeda) [inline addons/pharma/ACE_Medical_Treatment.hpp:296]: Why add maxRelief to medication that does not suppresses pain?

File: addons/pharma/ACE_Medical_Treatment.hpp [hunk 12]
```diff
@@ -157,6 +196,7 @@ class ACE_ADDON(Medical_Treatment) {
             maxDose = 4;
             incompatibleMedication[]= {};
             onOverDose = "";
+            maxRelief = 0.6;
```
Comment (MissHeda) [inline addons/pharma/ACE_Medical_Treatment.hpp:199]: Why add maxRelief to medication that does not suppresses pain?

File: addons/pharma/ACE_Medical_Treatment.hpp [hunk 13]
```diff
@@ -86,6 +122,7 @@ class ACE_ADDON(Medical_Treatment) {
             incompatibleMedication[] = {};
             viscosityChange = -10;
             onOverDose = "";
+            opioidRelief = 1.5;
```
Comment (MissHeda) [inline addons/pharma/ACE_Medical_Treatment.hpp:125]: Why add maxRelief to medication that does not suppresses pain?

**Pull request closed without merging**